dockstore/dockstore
# Enabling DOI generation for tags

**Description**
An alternative implementation to #5996.

Instead of an allowlist of GitHub orgs that should have DOI generation enabled, add a boolean property to individual workflows, app tools, notebooks and services (not tools) that controls whether DOIs should be generated.

This property defaults to false, and is currently only modifiable by an admin/curator API.

***Pros***
* Based on discussion in #5996, I think we need this property in any case, to handle both belated publishing of .dockstore.yml workflows, and for legacy workflows.
* To add workflows that should have DOI auto generation enabled, doesn't require logging into AWS (Secrets Manager update, force deployment) to change.
* When we're ready to move to opt-out, this property would still make sense
    * A .dockstore.yml value could set this property
    * The endpoint could be expanded to owners (or we drop the endpoint, and make the property editable via the existing updateWorkflow endpoint.

***Cons***
* Cannot set up in advance for a workflow that has not yet been created
* More cumbersome to configure for a GitHub organization, e.g., if we wanted to turn this feature on for an org, and an org has _n_ workflows, we'll have to make _n_ API requests. I could add or move the endpoint to handle this, but it seems like overkill.

**Review Instructions**
1. Set the property on a workflow
2. Create a tag on GitHub
3. Verify that a DOI was created
4. Create a tag on GitHub for a workflow that does not have this property set
5. Verify that a DOI was **not** created.

**Issue**
SEAB-6669

**Security and Privacy**

If there are any concerns that require extra attention from the security team, highlight them here and check the box when complete. 


e.g. Does this change...
* Any user data we collect, or data location?
* Access control, authentication or authorization?
* Encryption features?

Please make sure that you've checked the following before submitting your pull request. Thanks!
